lib/client/conntest/database/postgres.go

```diff
@@ -77,14 +77,13 @@ func (p *PostgresPinger) Ping(ctx context.Context, ping PingParams) error {
 	return nil
 }
 
-// IsConnectionRefusedError checks whether the error is of type invalid database user.
-// This can happen when the user doesn't exist.
+// IsConnectionRefusedError checks whether the error is of type connection refused.
 func (p *PostgresPinger) IsConnectionRefusedError(err error) bool {
 	if err == nil {
 		return false
 	}
 
-	return strings.Contains(err.Error(), "connection refused (SQLSTATE )")
+	return strings.Contains(err.Error(), "connection refused (SQLSTATE")
```

Comment:
Did you test this one?

From what I recall this extra space is required because the builder of this error string tries to add a code right after `SQLSTATE` but that's not always available.

I'm guessing it does something like `fmt.Sprintf("connection refused (SQLSTATE %s)", xyz)`
So, the space is always there even when there's no code.

Reply:
I deleted ` )` so it should match even when the code is blank. The difference is it will also match when the code is not blank

Reply:
And I did test it manually in the UI as well

Reply:
I'm sorry, I misread the diff :man_facepalming: 
All good then :+1: